src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketAwaitableEventArgs.cs

```diff
@@ -72,7 +78,7 @@ protected override void OnCompleted(SocketAsyncEventArgs _)
 
             if (continuation != null)
             {
-                _ioScheduler.Schedule(state => ((Action)state!)(), continuation);
+                _ioScheduler.Schedule(_pipeSchedulerDelegate, continuation);
```

Comment:
Are you sure this allocated before? I don't see any DisplayClasses getting new'ed up when I look at this in ILSpy. I'm pretty sure we would have noticed this in a memory profile if it was.